components/list/list-item-checkbox-mixin.js

```diff
@@ -86,7 +88,10 @@ export const ListItemCheckboxMixin = superclass => class extends SkeletonMixin(s
 	updated(changedProperties) {
 		super.updated(changedProperties);
 		if (!this._selectionProvider || !changedProperties.has('selectionInfo')) return;
-		this.selected = (this.selectionInfo.state === SelectionInfo.states.all);
+		// TODO: consider moving this to willUpdate
```

Comment:
Still considering?

Reply:
I kinda want to, but maybe not part of this PR since I'm nervous about the current change, as well as moving this. I can remove the comment if we prefer to have it.